keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,560 @@
+3. serviceaccount_jwks_fetch_requests_total
+
+- 1 and 2 can both be gauge metrics. 
+- Each will indicate how long it's been since the supported key set was last synced.
+- 3 can be a cumulatively increasing metric with `status` that will also enable deriving error rate.
+
+### Dependencies
+
+One new dependency will be introduced and it will only be required for clusters configured/opted-in via the `--service-account-signing-endpoint` flag.
+
+###### Does this feature depend on any specific services running in the cluster?
+
+The feature depends on a cluster level service that will implement [ExternalJWTSigner RPC](#externaljwtsigner-rpc) and will serve on UDS. each distribution will have their own version, so let's just address it as `ExternalJWTSigner`.
+
+- ExternalJWTSigner
+    - Usage description:
+      This service will act as a client to the external key management solution that a distribution will integrate with.
+
+      NOTE: this being a configurable/opt-in feature, the impact will only be seen on clusters that are using the feature.
+
+      - Impact of its outage on the feature:
+        - kube-apiserver will not be able to sign new service account JWTs.
+        - old JWTs will continue to be supported since we have an in cluster cache.
+        - refreshing supported key sets will not be possible.
+      - Impact of its degraded performance or high-error rates on the feature:
+        - service account token generation requests might require retries.
+        - issues when syncing support key sets might cause intermediate auth failures only if there are changes in supported key sets.
+
+### Scalability
+
+
+###### Will enabling / using this feature result in any new API calls?
+
+- Sign:
+  - This will be a call from kube-apiserver to an external service account JWT signer.
+  - The call will be made every time kube-apiserver receives a service account token request.
+  - It's a critical path to create pods and keep them running. 
+  - QPS will vary according to the cluster usage.
+  - ExternalJWTSigner's throughput will vary from distribution to distribution.
+
+- FetchKeys:
+  - This is again a call from kube-apiserver to the external service account JWT signer.
+  - It will be a periodic call to keep the OIDC provider in sync with the supported keys; Frequency will be decided by `suggested_next_fetch_at` returned by an external signer. 
+  - It will also be a dynamic call whenever kube-apiserver comes across a token signed by a key_id that it does not recognize.
+  - QPS will be rather low since:
+    - the signing keys shall not change very frequently.
+    - even a small value of `suggested_next_fetch_at` will result in less than 1 qps.
+  - ExternalJWTSigner's throughput will vary from distribution to distribution.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No new K8s api.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
```

Comment:
By cloud provider do we mean external signing source? I'm thinking of on-prem HSMs that might otherwise be used

Reply:
I Agree, while one of the motivation is to enable off cluster keys management with different cloud providers, there is no preventing of what externalSigners can integrate with.

will changed the language using might/can.